packages/mui-material/src/CircularProgress/CircularProgress.js

```diff
@@ -63,22 +71,38 @@ const CircularProgressRoot = styled('span', {
       styles[`color${capitalize(ownerState.color)}`],
     ];
   },
-})(
-  ({ ownerState, theme }) => ({
-    display: 'inline-block',
-    ...(ownerState.variant === 'determinate' && {
-      transition: theme.transitions.create('transform'),
-    }),
-    ...(ownerState.color !== 'inherit' && {
-      color: (theme.vars || theme).palette[ownerState.color].main,
-    }),
-  }),
-  ({ ownerState }) =>
-    ownerState.variant === 'indeterminate' &&
-    css`
-      animation: ${circularRotateKeyframe} 1.4s linear infinite;
-    `,
-);
+})(({ theme }) => ({
+  display: 'inline-block',
+  variants: [
+    {
+      props: {
+        variant: 'determinate',
+      },
+      style: {
+        transition: theme.transitions.create('transform'),
+      },
+    },
+    ...(typeof rotateAnimation !== 'string'
+      ? [
+          // For Styled-components v4+: https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/errors.md#12
+          {
+            props: {
+              variant: 'indeterminate',
+            },
+            style: rotateAnimation,
+          },
+        ]
+      : []),
```

Comment:
Do this to still support styled-components because it needs `css()`. When Pigment CSS is enabled, this styles will not be applied but inject directly via className (line: 208)

Reply:
Does

```
style: {
  animation: `${circularRotateKeyframe} 1.4s linear infinite`,
}
```

not work here?

Reply:
That works but you need to use inline style for the component so that it works for Pigment CSS whereas using `css""` produces a className that can be applied to the component directly.

Reply:
Ah, got it. let me try.

Reply:
Updated!